### ISSUE TITLE
Migration failed in Mysql 8, says "Invalid default value for `updated_at`"

### ISSUE THREAD

**Issue by wjh000123** [metabase/metabase#43985]
After I tried to upgrade to new version, the error occurred, which was:
 `Reason: liquibase.exception.DatabaseException: (conn=7301810) Invalid default value for 'updated_at' [Failed SQL: (1067) CREATE TABLE `metabase`.`cloud_migration` (`id` INT AUTO_INCREMENT NOT NULL COMMENT 'Unique ID', `external_id` LONGTEXT NOT NULL COMMENT 'Matching ID in Cloud for this migration', `upload_url` LONGTEXT NOT NULL COMMENT 'URL where the backup will be uploaded to', `state` VARCHAR(32) DEFAULT 'init' NOT NULL COMMENT 'Current state of the migration: init, setup, dump, upload, done, error, cancelled', `progress` INT DEFAULT 0 NOT NULL COMMENT 'Number between 0 to 100 representing progress as a percentage', `created_at` timestamp(6) NOT NULL COMMENT 'Timestamp when the config was inserted', `updated_at` timestamp(6) NOT NULL COMMENT 'Timestamp when the config was updated', CONSTRAINT `PK_CLOUD_MIGRATION` PRIMARY KEY (`id`)) COMMENT='Migrate to cloud directly from Metabase' ENGINE InnoDB CHARACTER SET utf8mb4 COLLATE utf8mb4_unicode_ci;]`

After some investigation, the root cause of mine issue is that the sql_mode in mysql 8 (you can check it by `show variables like 'sql_mode' ; `), it has `NO_ZERO_IN_DATE,NO_ZERO_DATE` by default, just delete these two items. Maybe the migration script shoud consider this situation.
_Originally posted by @wjh000123 in https://github.com/metabase/metabase/issues/40546#issuecomment-2159612060_

**Comment by qnkhuat** [comment]
I couldn't reproduce.

```
mysql> show variables like 'sql_mode';
+---------------+-----------------------------------------------------------------------------------------------------------------------+
| Variable_name | Value                                                                                                                 |
+---------------+-----------------------------------------------------------------------------------------------------------------------+
| sql_mode      | ONLY_FULL_GROUP_BY,STRICT_TRANS_TABLES,NO_ZERO_IN_DATE,NO_ZERO_DATE,ERROR_FOR_DIVISION_BY_ZERO,NO_ENGINE_SUBSTITUTION |
+---------------+-----------------------------------------------------------------------------------------------------------------------+
1 row in set (0.00 sec)
```

I have this and using MySQL 8.1.0 and migration works.

@wjh000123 could you provide more details about your setup? Maybe you have other custom settings?

**Comment by qnkhuat** [comment]
Which mysql version you're running and on which platform?

**Comment by paoliniluis** [comment]
yeah, me neither @wjh000123, can you specify exact version of MySQL you're using?

**Comment by paoliniluis** [comment]
setting it to "can't reproduce" as the user is either using a fork of MySQL/MariaDB or maybe a wrong version (like 5.x)

**Comment by WangDanpeng** [comment]
The issue seems to occur on the Alibaba Cloud database, and I have resolved it in a similar way. The problems tend to arise on Alibaba Cloud databases.
![image](https://github.com/metabase/metabase/assets/20870505/0e3549cd-7d8b-4ee0-9d13-d832d729364d)

**Comment by zhoujunhe** [comment]
这个问题很简单，启动metabase的时候把环境变量配置以下值，注意把IP，数据库，用户名和密码换成自己的，只要sql_mode没有NO_ZERO_IN_DATE,NO_ZERO_DATE就行，我里只使用了STRICT_TRANS_TABLES，大家用阿里云的应都会中文吧，这个在metabase官方文档还特意提了不支持阿里云的mysql，其实文档标记一下，碰到这问题让大家使用环境变量MB_DB_CONNECTION_URI就可以了。
```
MB_DB_CONNECTION_URI="jdbc:mysql://{IP}:3306/metabase?sessionVariables=sql_mode='STRICT_TRANS_TABLES'"
MB_DB_USER="username"
MB_DB_PASS="password"
```
